src/main/resources/init-hello.java.qute

```diff
@@ -3,6 +3,9 @@
 //DEPS {dep}
 {/for}
 {#if dependencies.isEmpty()}// //DEPS <dependency1> <dependency2>{/if}
+{#if compactSourceFiles}
+//JAVA 25+
```

Comment:
this is not symmetric fix - what if user chose a different java version?

the logic here is that if you run init with Java 25+ wehter available or chosen we generate for that - if anything it should generate //JAVA <versionchosen> but its not something we can do reliably.

Reply:
But isn't it true that it should be _at least_ Java 25? Otherwise it just won't work, right?

Reply:
> It also works on java22 with preview flags.

The `compactSourceFiles` is only `true` for Java 25+

> I'm just saying we also don't add java 17 when deps require java 17... Should we ?

Do we have templates that use deps that require Java 17? Then I would say yes we should. But only if we have templates where we have explicitly added those deps. I'm not suggesting we should do any detection or anything. Only straightforward things like have templates or examples where we _know_ the minimum version required.

> until now we haven't explicitly added //JAVA

Until now we haven't used any special Java features in our templates, haven't we? In some of our examples we do, but AFAIK those have //JAVA lines.